File: Checks/PWR013/example-omp.f90
! PWR013: Avoid copying unused variables to the GPU

subroutine example(A, B, C)
  implicit none
  integer, intent(in) :: A(:), B(:)
  integer, intent(inout) :: C(:)
  integer :: i

  !$omp target teams distribute parallel do schedule(auto) default(none) &
  !$omp& shared(A, B, C) private(i) map(to: A, B) map(tofrom: C)
  do i = 1, size(C, 1)
    C(i) = C(i) + A(i)
  end do
  !$omp end target teams distribute parallel do
end subroutine example

File: Checks/PWR013/example-omp.c
// PWR013: Avoid copying unused variables to the GPU

void example(double *A, double *B, double *C) {
  #pragma omp target teams distribute parallel for schedule(auto) shared(A, B) \
      map(to: A[0:100], B[0:100]) map(tofrom: C[0:100])
  for (int i = 0; i < 100; i++) {
    C[i] += A[i];
  }
}
